From 80e1e88e108b774c7aa7c201326f1a92cf1b7e35 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Wed, 26 Aug 2009 14:21:14 +0200
Subject: [PATCH 7/9] s390-tools-1.8.1-lszfcp-perf

---
 zconf/lszfcp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/zconf/lszfcp b/zconf/lszfcp
index d0bf84f..a872d05 100755
--- a/zconf/lszfcp
+++ b/zconf/lszfcp
@@ -233,7 +233,7 @@ show_devices()
 	# Differentiate old and new sysfs layout
 	if $FC_CLASS; then
 		SCSI_DEVICE_LIST=`ls -d \
-			$SYSFS/devices/css0/*/*/host*/rport*/target*/*/ \
+			$SYSFS/bus/ccw/drivers/zfcp/*/host*/rport*/target*/*/ \
 		       	2>/dev/null |grep -P '\d+:\d+:\d+:\d+'`
 	else
 		SCSI_DEVICE_LIST=`ls -d $SYSFS/devices/css0/*/*/host*/*/`
-- 
1.6.3.3

